### DISCUSSION TITLE
How to make a left join on a -to-One relation?

### DISCUSSION THREAD

**Discussion by TheVoid-0** [prisma/prisma#11801] [Q&A] [answered]
Hey there, is it possible to make a LEFT JOIN with WHERE on a to-one relation?

Example: I neet all items that have an event WHERE event.location = someLocationOR where this relation does not exists at all

```sql
SELECT i.* FROM items
LEFT JOIN events e ON e.id = items.event_id
WHERE e.location = ?
```

In prisma
```ts
prisma.items.findMany({
    where: {events: {
           location: someId
       }
   }
}) 
```
This query only brings the ITEMS that have a valid relation with events, but I ALSO want the items where the relation is non existing, meaning `event_id` in the `items` table can be either NULL or have an ID that does not matches with any event.

I'm able to make this sort of query with -to-many relations using `every`, but I couldn't figure it out how to make the same with -to-one relations.

I tried something like this

```ts
prisma.items.findMany({
    where:{
       OR: [events: { is: null }, {events: { location: someLocation} } ]
    }
})
```
But it didn't work.

Any help is apreciated

**Comment by nurul3101** [comment, marked as answer]
Hey @TheVoid-0 👋
Can you please share your schema file or just the `items` and `events` model?
I'll be able to take a look and could provide you with a solution by having a look at your schema file.
I am interested in the relation attributes of the models.

**Reply by TheVoid-0** [reply]
Hey there @nurul3101 thanks very much for answering and I'm sorry for the delay.

The situation I provided is an abstraction to my actual problem, but the relations are the same on the real world scenario I'm facing, here's what it looks like:

```prisma
model Items {
 id             Int        @id @default(autoincrement())
 eventId    Int?
 event       Events   @relation(fields: [eventId], references: [id]) 
}

model Events {
 id                Int        @id @default(autoincrement())  
 locationId   Int 
 item             Items?   // A event can have only one Item and one Item can have only one Event I.e One-to-One
}
```
Like I said in the original question, I want all Items where the event location id is in a list of ids OR where the relation with the event are non existent, since it is optional.

Thanks again for taking your time to help

**Reply by pantharshit00** [reply]
Hello @TheVoid-0, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

I think you can do the following query here:
```ts
  const data = await prisma.items.findMany({
    where: {
      OR: [
        {
          event: {
            id: {
              in: [1, 2, 3],
            },
          },
        },
        {
           eventId: null
        },
      ],
    },
  });
```

**Reply by TheVoid-0** [reply]
No problem @pantharshit00, I don't actually remember where this was necessary but I think it was solved very similarly, anyway I wanted to know if a left join was possible but looks like that's not suported on -to-one relations.

**Comment by applemate** [comment]
this is overly complicated. sometime raw sql is just much much faster to write. And prisma syntax is really ugly with all that braces and curly.

**Reply by halwa343741** [reply]
agree, raw sql is the full power of RDBMS.. nothing can replace it..

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
